sdk/core/core-client/src/serializer.ts

```diff
@@ -239,7 +239,7 @@ class SerializerImpl implements Serializer {
         responseBody = [];
       }
       // specifically check for undefined as default value can be a falsey value `0, "", false, null`
-      if (mapper.defaultValue !== undefined) {
+      if (mapper.required && mapper.defaultValue !== undefined) {
```

Comment:
I feel like this is a pretty big change in runtime behavior. To ensure we didn't break we'd have to audit every use of `defaultValue` in a mapper for all SDKs we've shipped.